lib/lotus/model/adapters/memory/query.rb

```diff
@@ -554,6 +555,40 @@ def group
             raise NotImplementedError
           end
 
+          # Preload a given association into root's aggregation
+          # This should be implemented inside Repository in a class method. See example below.
+          #
+          # @example
+          #
+          #     mapping do
+          #       collections :users do
+          #         entity User
+          #         attribute :id, Integer
+          #         association :articles, [Article], foreign_key: :user_id, collection: :articles
+          #       end
+          #
+          #       collections :articles do
+          #         entity Article
+          #         attribute :id, Integer
+          #         attribute :user_id, Integer
+          #         association :user, [User], foreign_key: :id, collection: :articles
```

Comment:
I think this example should be `User` instead `[User]`because we have a singular association